server/backend/database/memory/database.go

```diff
@@ -247,7 +247,19 @@ func (d *DB) listProjectInfos(
 	for i := 0; i < pageSize; i++ {
 		raw := iter.Next()
 		if raw == nil {
-			break
+			iter, err = txn.LowerBound(
```

Comment:
Does this query repeats for `pageSize` - `fetched projects size`?
We might need to improve these codes.

Reply:
I have made the adjustments according to your feedback. I have refactored the code to use common logic without re-invoking `raw = iter.Next()` within the if statement. However, the impact of the improvement seems to be limited. If you have any suggestions or opinions, please share it.